numpy/core/_methods.py

```diff
@@ -189,8 +189,25 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     # Note that x may not be inexact and that we need it to be an array,
     # not a scalar.
     x = asanyarray(arr - arrmean)
+
+    # Complex types to -> (2,)float view for fast-path
+    _complex_to_float = {
+        nt.complex64 : nt.float32,
+        nt.complex128 : nt.float64,
+        nt.clongdouble: nt.longdouble
```

Comment:
If you make the keys `np.dtype` instances, and crucially reverse the order, then you can avoid the `isnative` check below.

Reply:
Reversing is necessary so that when double and longdouble are equivalent, the latter, double, wins.

Reply:
I don't understand the comment on ordering; particularly the second comment. Also, I thought that Python dictionaries didn't make any guarantees about ordering. Would this need to be an OrderedDict?

Reply:
Dictionary literals are evaluated sequentially. double and long double are considered equal on Windows, you want double to take precedence.

Reply:
Apologies for the lack of explanation, will expand when not on Mobile.

Reply:
Perhaps best to keep things explicit:
```python
_complex_to_float = {
    np.dtype(np.csingle): np.dtype(np.single),
    np.dtype(np.cdouble): np.dtype(np.double),
}
# on windows these are equivalent, and we want double to take precedence
if np.dtype(np.longdouble) != np.dtype(np.double):
    _complex_to_float.update({
    	np.dtype(np.clongdouble): np.dtype(np.longdouble),
    })
# add the reverse-endian types
_complex_to_float.update({
	k.newbyteorder(): v.newbyteorder()
	for k, v in _complex_to_float.items()
})
```

---

The remark I was making above is that:
```python
d = {
    1: "a",
    2: "b",
    1: "c"
}
```
is treated as equivalent to
```python
d = {}
d[1] = "a"
d[2] = "b"
d[1] = "c"
```
eg when repeated, the last repetition wins.

Reply:
I see now, thanks for the explanation - one thing however: I understood from @seberg 's comment that non-native types are to be excluded, so we wouldn't want to add the reverse byteorders to the fast-path dictionary

Reply:
I think @seberg meant that viewing a big-endian `complex double` as a little-endian `double` was unsafe. As long as you match the endianness before and after the view, everything will be fine.

Reply:
Thanks for the clarification :+1:

Reply:
One issue with the concrete dtype instances is that attempting to import `dtype` from `numpy.core` or the full namespace (`import numpy as np`) results in a circular import in `numpy/core/_methods.py`. Is there a way to get concrete datatype instances using only `numpy.core.numerictypes`?

Reply:
`np.core.numerictypes.dtype`?

Reply:
Thanks, I don't know why I had such a hard time chasing that down :expressionless: